include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -812,3 +813,59 @@ def JoinOp : Handshake_Op<"join", [
 
 def I4 : I<4>;
 def I4Attr : SignlessIntegerAttrBase<I4, "4-bit integer attribute">;
+
+// TODO(Dinistro): add support for this in the handshake-runner
+def UnpackOp : Handshake_Op<"unpack", [
+  TypesMatchWith<"result types match element types of 'tuple'",
+                     "input", "results",
+                     "$_self.cast<TupleType>().getTypes()">
+]> {
+  let summary = "unpacks a tuple";
+  let description = [{
+    The `unpack` operation assignes each value of a tuple to a separate
+    value for further processing. The number of results corresponds
+    to the number of tuple elements.
```

Comment:
For both unpack and pack, i think there should be a note about expected behaviour, i.e., that control logic for these mimick that of fork/joins.